test/functional/p2p_filter.py

```diff
@@ -64,95 +91,144 @@ def set_test_params(self):
-            filter_node.wait_for_tx(txid)
+            filter_peer.wait_for_tx(txid)
 
         self.log.info('Check that request for filtered blocks is ignored if no filter is set')
-        filter_node.merkleblock_received = False
-        filter_node.tx_received = False
+        filter_peer.merkleblock_received = False
+        filter_peer.tx_received = False
         with self.nodes[0].assert_debug_log(expected_msgs=['received getdata']):
             block_hash = self.nodes[0].generatetoaddress(1, self.nodes[0].getnewaddress())[0]
-            filter_node.wait_for_inv([CInv(MSG_BLOCK, int(block_hash, 16))])
-            filter_node.sync_with_ping()
-            assert not filter_node.merkleblock_received
-            assert not filter_node.tx_received
+            filter_peer.wait_for_inv([CInv(MSG_BLOCK, int(block_hash, 16))])
+            filter_peer.sync_with_ping()
+            assert not filter_peer.merkleblock_received
+            assert not filter_peer.tx_received
 
         self.log.info('Check that sending "filteradd" if no filter is set is treated as misbehavior')
         with self.nodes[0].assert_debug_log(['Misbehaving']):
-            filter_node.send_and_ping(msg_filteradd(data=b'letsmisbehave'))
+            filter_peer.send_and_ping(msg_filteradd(data=b'letsmisbehave'))
 
         self.log.info("Check that division-by-zero remote crash bug [CVE-2013-5700] is fixed")
-        filter_node.send_and_ping(msg_filterload(data=b'', nHashFuncs=1))
-        filter_node.send_and_ping(msg_filteradd(data=b'letstrytocrashthisnode'))
+        filter_peer.send_and_ping(msg_filterload(data=b'', nHashFuncs=1))
+        filter_peer.send_and_ping(msg_filteradd(data=b'letstrytocrashthisnode'))
+        self.nodes[0].disconnect_p2ps()
 
+    def run_test(self):
+        filter_peer = self.nodes[0].add_p2p_connection(P2PBloomFilter())
+        self.log.info('Test filter size limits')
+        self.test_size_limits(filter_peer)
+
+        self.log.info('Test BIP 37 for a node with fRelay = True (default)')
+        self.test_filter(filter_peer)
+        self.nodes[0].disconnect_p2ps()
+
+        self.log.info('Test BIP 37 for a node with fRelay = False')
+        # Add peer but do not send version yet
+        filter_peer_without_nrelay = self.nodes[0].add_p2p_connection(P2PBloomFilter(), send_version=False, wait_for_verack=False)
+        # Send version with fRelay=False
+        filter_peer_without_nrelay.wait_until(lambda: filter_peer_without_nrelay.is_connected, timeout=10)
+        version_without_fRelay = msg_version()
+        version_without_fRelay.nRelay = 0
+        filter_peer_without_nrelay.send_message(version_without_fRelay)
+        filter_peer_without_nrelay.wait_for_verack()
+        assert not self.nodes[0].getpeerinfo()[0]['relaytxes']
+        self.test_frelay_false(filter_peer_without_nrelay)
+        self.test_filter(filter_peer_without_nrelay)
+
+        self.log.info('Test msg_mempool')
```

Comment:
497a6193 Is this logging (or the one at the start of `test_msg_mempool()`) redundant?